src/entries/popup/pages/settings/walletsAndKeys/hardwareWalletWipePrompt.tsx

```diff
@@ -0,0 +1,174 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Address } from 'wagmi';
+
+import { i18n } from '~/core/languages';
+import { appSessionsStore, useCurrentAddressStore } from '~/core/state';
+import { hiddenWalletsStore } from '~/core/state/hiddenWallets';
+import { walletBackupsStore } from '~/core/state/walletBackups';
+import { walletNamesStore } from '~/core/state/walletNames';
+import { KeychainWallet } from '~/core/types/keychainTypes';
+import { getSettingWallets } from '~/core/utils/settings';
+import {
+  Box,
+  Button,
+  Column,
+  Columns,
+  Inset,
+  Row,
+  Rows,
+  Separator,
+  Text,
+} from '~/design-system';
+import { Prompt } from '~/design-system/components/Prompt/Prompt';
+import { getAccounts, remove, wipe } from '~/entries/popup/handlers/wallet';
+import { useRainbowNavigate } from '~/entries/popup/hooks/useRainbowNavigate';
+import { ROUTES } from '~/entries/popup/urls';
+
+const { deleteWalletName } = walletNamesStore.getState();
+const { deleteWalletBackup } = walletBackupsStore.getState();
+const { removeAddressSessions } = appSessionsStore.getState();
+const { unhideWallet } = hiddenWalletsStore.getState();
+
+async function removeWallet(address: Address) {
+  await remove(address);
+  unhideWallet({ address }); // unhide so if it's readded later, it's not hidden
+  deleteWalletName({ address });
+  deleteWalletBackup({ address });
+  removeAddressSessions({ address });
+}
+
+export const HardwareWalletWipePrompt = ({
+  show,
+  onClose,
+}: {
+  show: boolean;
+  onClose: () => void;
+}) => {
+  const { currentAddress, setCurrentAddress } = useCurrentAddressStore();
+  const navigate = useRainbowNavigate();
+  const [wallet, setWallet] = useState<KeychainWallet | null>();
+
+  useEffect(() => {
+    const fetchWallets = async () => {
+      const walletFromKeychain = await getSettingWallets();
+      console.log(walletFromKeychain);
+      setWallet(walletFromKeychain);
+    };
+
+    fetchWallets();
+  }, [currentAddress]);
+
+  const handleDeleteHardwareWalletGroup = useCallback(async () => {
+    const groupAccounts = (await getSettingWallets()).accounts;
+    try {
+      await Promise.all(groupAccounts.map(removeWallet));
+
+      if (groupAccounts.includes(currentAddress)) {
+        const allAccounts = await getAccounts();
+
+        if (allAccounts.length === 0) {
+          await wipe();
+          navigate(ROUTES.WELCOME);
+          return;
+        }
+
+        const { hiddenWallets } = hiddenWalletsStore.getState();
+        const visibleWallets = allAccounts.filter((acc) => !hiddenWallets[acc]);
+        // if no more visible wallets, force unhide one and set as current
+        if (visibleWallets.length === 0) {
+          unhideWallet({ address: allAccounts[0] });
```

Comment:
shouldn't this use hiddenWallets[0] instead?

Reply:
I think in this case they would return the same thing. There may be an edge case where what you say needs to be done though. I think i would have to do something like:
```suggestion
          unhideWallet({
            address: Object.keys(hiddenWallets)[0] as `0x${string}`,
          });
```

Reply:
but if visibleWallets.length === 0 then unhiding allAccounts[0] i think does the same thing. i may be confused tho.

Reply:
that new change lgtm, is nice to maintain an order of names 👌  unhiding hidden wallets makes more sense than unhiding visible wallets